extras/Build/CMake/JUCEUtils.cmake

```
@@ -1337,6 +1347,40 @@ function(_juce_set_plugin_target_properties shared_code_target kind)
         endif()
 
         _juce_copy_after_build(${shared_code_target} ${target_name} "${output_path}" JUCE_VST3_COPY_DIR)
+    elseif(kind STREQUAL "LV2")
+        set_target_properties(${target_name} PROPERTIES
+            BUNDLE_EXTENSION lv2
+            PREFIX ""
+            SUFFIX .lv2
+            BUNDLE TRUE
+            XCODE_ATTRIBUTE_WRAPPER_EXTENSION lv2
+            XCODE_ATTRIBUTE_LIBRARY_STYLE Bundle
+            XCODE_ATTRIBUTE_GENERATE_PKGINFO_FILE YES)
+
+        _juce_create_windows_package(${shared_code_target} ${target_name} lv2 "" x86-win x86_64-win)
+
+        set(output_path "${products_folder}/${product_name}.lv2")
+
+        if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+            # On linux we assume that the output arch is the same as the that of the host platform
+            set(is_platform_x64 $<EQUAL:${CMAKE_SIZEOF_VOID_P},8>)
+            set(arch_string $<IF:${is_platform_x64},x86_64,i386>)
+
+            set_target_properties(${target_name} PROPERTIES
+                SUFFIX .so
+                LIBRARY_OUTPUT_DIRECTORY "${output_path}")
+        endif()
+
+        # generate .ttl files
+        add_executable(lv2_ttl_generator ${JUCE_SOURCE_DIR}/extras/Build/lv2_ttl_generator/lv2_ttl_generator.c)
+        target_link_libraries(lv2_ttl_generator dl)
+        add_custom_command(TARGET ${target_name} POST_BUILD
+            COMMAND lv2_ttl_generator "${output_path}/${shared_code_target}.so"
+            DEPENDS ${target_name} lv2_ttl_generator
+            WORKING_DIRECTORY "${products_folder}/${product_name}.lv2/"
+            VERBATIM)
```

Can I instead suggest:

```suggestion
        add_custom_command(TARGET ${target_name} POST_BUILD
            COMMAND lv2_ttl_generator "$<TARGET_FILE:${target_name}>"
            COMMENT "Generating LV2 Turtle manifest files for ${target_name}"
            WORKING_DIRECTORY "${products_folder}/${product_name}.lv2/"
            VERBATIM)
```

The CMake [docs](https://cmake.org/cmake/help/latest/command/add_custom_command.html) say:

> If `COMMAND` specifies an executable target name (created by the `add_executable()` command), it will automatically be replaced by the location of the executable created at build time...

So just writing `COMMAND lv2_ttl_generator` will be sufficient to 1) add an automatic build dependency on the target `lv2_ttl_generator` and 2) run it in a cross-platform way on `$<TARGET_FILE:${target_name}>`.

Note that if you decide to change the output directory then you'll need to change the suggested `WORKING_DIRECTORY` here, too.

---

Okay yeah, that makes sense! Let me just give it a try first...